### DISCUSSION TITLE
Can I use the same model or enum for emitting both openapi and protobuf?

### DISCUSSION THREAD

**Discussion by flappyBug** [microsoft/typespec#4043] [Q&A]
Hi, there.  I'm new to typespec and I'd like to use typespec for defining messaging protos for my project. I'm using restful api for message between my frontend and my api server, and protobuf for my api server and other microservices.

I'm wondering if I can reuse models and enums to emitting different representations for openapi and protobuf. For example:

```tsp
enum TradeDirection {
    @Protobuf.field(0)
    @Restful.field('buy')
    BUY;
    @Protobuf.field(1)
    @Restful.field('sell')
    SELL;
}

model Task {
    @Protobuf.field(1)
    direction: TradeDirection;
}

namespace HTTP {
    interface Service {
        createTask(Task) 
    }
}

namespace Protobuf {
    model TaskWithUserID extends Task {
        @Protobuf.field(2) userId: string;
    }
    interface Service {
        createTask(TaskWithUserID) 
    }
}
```

**Comment by timotheeguerin** [comment]
you can definitely do that, TypeSpec was designed with this scenario in mind. You do have to be mindful of course of how certain protocol might require certain structure(specially around operations) so you don't affect others but otherwise it should definitely work.

We have this example here showcasing multiple emitters https://typespec.io/multi-protocol

**Reply by flappyBug** [reply]
Thanks for the link. A more complex example can be helpful. For example, Can I have the same enum that serializes to number in profobuf, and serializes to string in openapi. Can I extends the base model for the specific proto.

**Comment by BenjaminTitmusGMSL** [comment]
I have a specific question related to this. If I want to use the `@statusCode` decorator on a model then this doesn't work with `@typespec/protobuf` unless I am using it wrong:

```
model Response {
  @statusCode _: 200;
}
```

This results in an error that I haven't specified a field for this. But I don't really want to because really the status code should be a decorator on the model, not a field within the model. In any case adding a `@field` decorator doesn't work because I can't give a single value for a field I need to give a type. I can do the following if I have a range:

```
model Response {
  @statusCode
  @minValue(200)
  @maxValue(201)
  @field(1)
  code: int32;
}
```

Even in this case it will still emit the `code` field which I don't want.
